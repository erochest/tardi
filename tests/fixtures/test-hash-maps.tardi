
uses: std/hashmaps
uses: std/io
uses: std/strings
uses: std/vectors

"<hashmap> : " print
<hashmap> .

">hashmap : " print
{ { 'a' 1 } { 'b' 2 } { 'c' 3 } } >hashmap
is-hashmap? .

"H{ : " print
H{ { 'a' 4 } { 'b' 5 } { 'c' 6 } }
{ { 'a' 4 } { 'b' 5 } { 'c' 6 } } >hashmap
== .

"== : " print
H{ { 'a' 4 } { 'b' 5 } { 'c' 6 } }
H{ { 'a' 4 } { 'b' 5 }  } == .

">string : " print
H{ { "given-name" "Trillian" } { "surname" "McMillan" } } strings/>string .

"length : " print
H{ } hashmaps/length .
H{ { 1 2 } } hashmaps/length .
H{ { 1 2 } { 2 3 } { 3 4 } { 4 5 } }
dup hashmaps/length .

">vector : " print
dup >vector dup sort! .

"keys : " print
dup keys dup sort! .

"values : " print
dup values dup sort! .

"in? : " print
1 over hashmaps/in? .
9 over hashmaps/in? .

"get : " print
2 over get . .
9 over get . .

"add! : " print
H{ { 'a' 1 } { 'b' 2 } { 'd' 4 } { 'e' 5 } }
{ 'c' 3 } over add!
dup >vector dup sort! .

"set! : " print
'd' 9 pick set!
>vector dup sort! .

"each :" println
{ } { } pick [
  pick push!
  pick push!
] hashmaps/each
dup sort!
[ dup sort! ] dip
"  values: " print .
"  keys  : " print .

// TODO: map        ( lambda:( v -- v' ) hashmap -- hashmap )
// TODO: empty?
