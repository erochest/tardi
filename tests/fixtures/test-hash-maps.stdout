<hashmap> : H{ }
>hashmap : #t
H{ : #t
== : #f
>string : "H{ { \"given-name\" \"Trillian\" } { \"surname\" \"McMillan\" } }"
length : 0
1
4
>vector : { { 1 2 } { 2 3 } { 3 4 } { 4 5 } }
keys : { 1 2 3 4 }
values : { 2 3 4 5 }
in? : #t
#f
get : #t
3
#f
#f
add! : { { 'a' 1 } { 'b' 2 } { 'c' 3 } { 'd' 4 } { 'e' 5 } }
set! : { { 'a' 1 } { 'b' 2 } { 'c' 3 } { 'd' 9 } { 'e' 5 } }
each :
  values: { 2 3 4 5 }
  keys  : { 1 2 3 4 }
